vector/src/main/scala/geotrellis/vector/reproject/Reproject.scala

```diff
@@ -100,7 +100,9 @@ object Reproject {
       val length = sqrt(pow(x0 - x1, 2) + pow(y0 - y1, 2))
 
       val p2 = m -> (x2, y2)
-      if (deflect / length < relError) {
+      if (deflect.isNaN) {
```

Comment:
I recall we did a bit of benchmarking and found that `double.isNaN` was surprisingly and significantly slower than `java.lang.Double.isNaN` which is why we chose to use that in `isNoData` macro: https://github.com/locationtech/geotrellis/blob/c5094d0437e3d36436fc2d40975a1e99ec775401/macros/src/main/scala/geotrellis/macros/NoDataMacros.scala#L44

This is looks like tight enough loop that we should use that.

Reply:
Totally forgot about it, I even benchmarked it, you're correct.

Reply:
Totally forgot about it, I even benchmarked it, you're correct.